### ISSUE TITLE
ArgumentError: invalid configuration option `:use_iam_profile'

### ISSUE THREAD

**Issue by ericvautour-quandl** [iHiD/propono#89]
Hello All,

I am receiving the following error when attempting to publish a message.  Was use_iam_profile deprecated when FOG was removed?

CALL:
```
client = Propono::Client.new do |config|
  config.use_iam_profile = true
  config.queue_region = "us-east-1"
  config.application_name = "PermissionServiceBus"
  config.logger = Rails.logger
end

client.publish('topic1', 'A Simple Message')
```

ERROR:
```
ArgumentError: invalid configuration option `:use_iam_profile'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/aws-sdk-core-3.21.2/lib/seahorse/client/configuration.rb:166:in `rescue in block in apply_options'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/aws-sdk-core-3.21.2/lib/seahorse/client/configuration.rb:162:in `block in apply_options'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/aws-sdk-core-3.21.2/lib/seahorse/client/configuration.rb:161:in `each'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/aws-sdk-core-3.21.2/lib/seahorse/client/configuration.rb:161:in `apply_options'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/aws-sdk-core-3.21.2/lib/seahorse/client/configuration.rb:149:in `build!'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/aws-sdk-core-3.21.2/lib/seahorse/client/base.rb:62:in `build_config'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/aws-sdk-core-3.21.2/lib/seahorse/client/base.rb:19:in `initialize'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/aws-sdk-sns-1.2.0/lib/aws-sdk-sns/client.rb:143:in `initialize'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/aws-sdk-core-3.21.2/lib/seahorse/client/base.rb:99:in `new'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/propono-2.1.0/lib/propono/components/aws_client.rb:71:in `sns_client'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/propono-2.1.0/lib/propono/components/aws_client.rb:26:in `create_topic'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/propono-2.1.0/lib/propono/services/publisher.rb:41:in `publish_syncronously'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/propono-2.1.0/lib/propono/services/publisher.rb:30:in `publish'
from .rvm/gems/ruby-2.3.4@wikiposit/gems/propono-2.1.0/lib/propono/services/publisher.rb:9:in `publish'
```

**Comment by iHiD** [comment]
Propono still supports it. I'm wondering if the AWS gem has changed in a recent release. We have this working in production using `aws-sdk-core (3.7.0)` and `aws-sdk-sns (1.1.0)`. Could you try locking your Gemfile to those and seeing if that fixes it for you please?

I'm going to be snowed under for the next week or two, but that might give you a fix for now.

**Comment by dougal** [comment]
Closed by #102.

Hashes can now be passed to the SQS and SNS client initialisation. This allows support for the recommended `Aws::InstanceProfileCredentials` approach to using IAM profiles.

Example:

```
role_credentials = Aws::AssumeRoleCredentials.new(
  client: Aws::STS::Client.new,
  role_arn: "linked::account::arn",
  role_session_name: "session-name"
)

client = Propono::Client.new do |config|
  config.aws_options = {
    credentials: role_credentials       
  }
end
```
